components/Button/index.tsx

```diff
@@ -28,7 +28,7 @@ const Button: FC<ButtonProps> = ({
 }) => {
   return (
     <button type={type} className={`${className} ${styles.button}`} onClick={onClick}>
-      {withIcon && (
+      {withIcon && iconName && (
```

Comment:
In this case what about removing within prop and using icon name as boolean?

```suggestion
      {!!iconName && (
```

Reply:
Done.